contracts/andromeda_gumball/src/contract.rs

```diff
@@ -0,0 +1,881 @@
+use crate::state::{CW721_CONTRACT, LIST, STATE, STATUS};
+use ado_base::ADOContract;
+use andromeda_protocol::gumball::{LatestRandomResponse, State};
+use andromeda_protocol::{
+    cw721::{ExecuteMsg as Cw721ExecuteMsg, MintMsg, TokenExtension},
+    gumball::{
+        ExecuteMsg, InstantiateMsg, NumberOfNFTsResponse, QueryMsg, StateResponse, StatusResponse,
+    },
+};
+use common::{
+    ado_base::{recipient::Recipient, InstantiateMsg as BaseInstantiateMsg},
+    encode_binary,
+    error::ContractError,
+    require,
+};
+use cosmwasm_std::{attr, entry_point, Binary};
+use cosmwasm_std::{
+    Coin, CosmosMsg, Deps, DepsMut, Env, MessageInfo, QueryRequest, Response, Uint128, WasmMsg,
+    WasmQuery,
+};
+use cw2::set_contract_version;
+const CONTRACT_NAME: &str = "crates.io:andromeda_gumball";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+const TERRAND_ADDRESS_TESTNET: &str = "terra1a62jxn3hh54fa5slan4dkd7u6v4nzgz3pjhygm";
```

Comment:
Should we be hardcoding this?

Reply:
This should probably be provided in the instantiation message

Reply:
It's now part of the instantiation message, saved in the Item `RANDOMNESS`.